toolchain/lowering/lowering_context.cpp

```diff
@@ -165,6 +165,23 @@ auto LoweringContext::BuildType(SemanticsNodeId node_id) -> llvm::Type* {
       return llvm::StructType::create(*llvm_context_, subtypes,
                                       "StructLiteralType");
     }
+    case SemanticsNodeKind::TupleType: {
+      auto refs = semantics_ir_->GetTypeBlock(node.GetAsTupleType());
+      llvm::SmallVector<llvm::Type*> subtypes;
+      subtypes.reserve(refs.size());
+      for (auto ref_id : refs) {
+        if (ref_id == SemanticsTypeId::TypeType) {
+          // TODO: Handle TypeType. Currently using a place holder
+          // but this is just a workaround not an actual solution.
+          subtypes.push_back(llvm::Type::getInt1Ty(*llvm_context_));
+        } else {
+          subtypes.push_back(GetType(ref_id));
+        }
+        CARBON_CHECK(ref_id.index < SemanticsBuiltinKind::ValidCount) << ref_id;
```

Comment:
This `CHECK` looks incorrect to me: it's comparing a `SemanticsTypeId` to a `SemanticsBuiltinKind`, and seems to be ensuring that we only see tuple types containing builtin types, which won't be true in general. It might be reasonable to check that we don't see `SemanticsTypeId::Error` here, but it looks like that'll check-fail in `GetType`, so it seems like that case is already handled and we can just remove this `CHECK`.